src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+        private const string RegistrationName = "OnnxTransform";
+
+        internal const string Summary = "Transforms the data using the Onnx model.";
+        internal const string UserName = "OnnxTransform";
+        internal const string ShortName = "Onnx";
+        internal const string LoaderSignature = "OnnxTransform";
+
+        public readonly string[] Inputs;
+        public readonly string[] Outputs;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "ONNXSCOR",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        // Factory method for SignatureDataTransform
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            return new OnnxTransform(env, args).MakeDataTransform(input);
+        }
+
+        // Factory method for SignatureLoadDataTransform
+        public static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+            => Create(env, ctx).MakeDataTransform(input);
+
+        // Factory method for SignatureLoadModel.
+        public static OnnxTransform Create(IHostEnvironment env, ModelLoadContext ctx)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel(GetVersionInfo());
+
+            byte[] modelBytes = null;
+            if (!ctx.TryLoadBinaryStream("OnnxModel", r => modelBytes = r.ReadByteArray()))
+                throw env.ExceptDecode();
+
+            var inputColumn = ctx.LoadNonEmptyString();
+            var outputColumn = ctx.LoadNonEmptyString();
+            var args = new Arguments() { InputColumn = inputColumn, OutputColumn = outputColumn };
+
+            return new OnnxTransform(env, args, modelBytes);
+        }
+
+        // Factory method for SignatureLoadRowMapper.
+        public static IRowMapper Create(IHostEnvironment env, ModelLoadContext ctx, ISchema inputSchema)
+            => Create(env, ctx).MakeRowMapper(inputSchema);
+
+        private OnnxTransform(IHostEnvironment env, Arguments args, byte[] modelBytes = null)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            _host = env.Register(RegistrationName);
+            _host.CheckValue(args, nameof(args));
+            _host.CheckNonWhiteSpace(args.InputColumn, nameof(args.InputColumn));
+            _host.CheckNonWhiteSpace(args.OutputColumn, nameof(args.OutputColumn));
+
+            if (modelBytes == null)
+            {
+                _host.CheckNonWhiteSpace(args.ModelFile, nameof(args.ModelFile));
+                _host.CheckUserArg(File.Exists(args.ModelFile), nameof(args.ModelFile));
+                _model = new OnnxModel(args.ModelFile);
```

Comment:
>_model [](start = 16, length = 6)

is there a memory limitation of keeping everything in memory ?  i would presume some Onnx models to be quite large

Reply:
This API is what is supported in Sonoma currently. There's no alternative at this point. This can be a design discussion for the Lotus# library. Making as resolved.

---
In reply to: [219921638](https://github.com/dotnet/machinelearning/pull/942#discussion_r219921638) [](ancestors = 219921638)